src/configuration.c

```diff
@@ -112,55 +159,36 @@ bool read_config_from_file(char *config_file, struct configuration *config)
                                         return false;
                                 }
                         } else {
-                                fprintf(stderr, "ERR: missing key with string value: %s\n", config_key_str[i]);
-                                return false;
+                                config->strValues[i] = strdup(config_str_default[i]);
                         }
+
                 }
-                for (int i = CONF_INT_MIN + 1; i < CONF_INT_MAX; i++) {
+
+                for (int i = CONF_INT_MIN; i < CONF_INT_MAX; i++) {
                         char *ptr;
                         ptr = nc_hashmap_get(nc_hashmap_get(keyfile, "settings"), config_key_int[i]);
                         if (ptr) {
                                 config->intValues[i] = strtoll(ptr, NULL, 10);
```

Comment:
Not part of the patch, however, for "strtoll"  the calling program should set errno to 0 before the call, and then determine if an error occurred by checking whether errno has a nonzero value after the call.